boards/arm/zybo/zybo.dts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2022 Henrik Brix Andersen <henrik@brixandersen.dk>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <xilinx/zynq7000.dtsi>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+
+#include "zybo-pinctrl.dtsi"
+
+/ {
+	model = "Digilent Zybo board";
+	compatible = "digilent,zynq-zybo", "xlnx,zynq-7000";
+
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &uart1;
+		zephyr,shell-uart = &uart1;
+		zephyr,ocm = &ocm_high;
+	};
+
+	aliases {
+		led0 = &ld_mio;
+		sw0 = &btn4;
+		sw1 = &btn5;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a9";
+			reg = <0>;
+		};
+
+		cpu1: cpu@1 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a9";
+			reg = <1>;
+		};
+	};
+
+	sram0: memory@0 {
+		compatible = "mmio-sram";
+		reg = <0x00000000 DT_SIZE_M(512)>;
+	};
```

Comment:
question: why is this defined at board level? is it because of the nature of the SoC (FPGA)?

Reply:
> question: why is this defined at board level? is it because of the nature of the SoC (FPGA)?

Regarding the RAM: the Zynq has an integrated DDR memory controller, but other than some on-chip memory, no internal RAM. So the RAM size depends on what is actually soldered onto the respective board. The same applies to flash memory.

Regarding the CPUs: the Zynq is available in two general flavors: XC7Zxxx, which has two A9 cores, and XC7ZxxxS, which only has one core. Several sub-models, which differ in I/O pin count and FPGA type / logic cell count (Artix-7 for the smaller models, Kintex-7 for the larger models), exist for both of these. However, despite a varying set of actually available peripherals (due to I/O pin constraints) and programmable logic space, all those chips always have the same set of on-board peripherals. Therefore, the CPU nodes were moved to the board level, initially for qemu_cortex_a9, in order to:
- not to have two dts files with what may eventually become a sizeable set of supported peripherals, whose only difference is the 1 vs. 2 CPU core count,
- allow selective SMP / single-core operation with the same board by means of overlays or a similar solution (comp. qemu_cortex_a53 which optionally has an _smp variant).
Another minor factor for this move was the consistency between the device tree used for building Zephyr and the device tree provided to QEMU for the qemu_cortex_a9 target. The latter effectively specifies a single-core Zynq, as QEMU is acting up if a 2nd core is present but not taken out of the boot loop by the OS due to a lack of SMP capability.
All this was introduced in #40964.

Reply:
As @ibirnbaum wrote, the SDRAM is external and thus board-specific. I would like to move the CPU nodes into the SoC DTS file (we can easily handle single- vs. dual-core in those, even from overlays) along with some other clean-ups in a later PR.